core/src/main/kotlin/com/tschuchort/compiletesting/AbstractKotlinCompilation.kt

```diff
@@ -239,23 +241,25 @@ abstract class AbstractKotlinCompilation<A : CommonCompilerArguments> internal c
     }
 
     protected fun getResourcesPath(): String {
-        val resourceName = "META-INF/services/org.jetbrains.kotlin.compiler.plugin.ComponentRegistrar"
         return this::class.java.classLoader.getResources(resourceName)
             .asSequence()
-            .mapNotNull { url ->
-                val uri = URI.create(url.toString().removeSuffix("/$resourceName"))
-                when (uri.scheme) {
-                    "jar" -> Paths.get(URI.create(uri.schemeSpecificPart.removeSuffix("!")))
-                    "file" -> Paths.get(uri)
-                    else -> return@mapNotNull null
-                }.toAbsolutePath()
-            }
+            .mapNotNull { url -> urlToResourcePath(url) }
             .find { resourcesPath ->
                 ServiceLoaderLite.findImplementations(ComponentRegistrar::class.java, listOf(resourcesPath.toFile()))
                     .any { implementation -> implementation == MainComponentRegistrar::class.java.name }
             }?.toString() ?: throw AssertionError("Could not get path to ComponentRegistrar service from META-INF")
     }
 
+    internal fun urlToResourcePath(url: URL): Path? {
```

Comment:
Can you doc this function for posterity?

Reply:
85b5a7c